### PR TITLE
Increase coverage

### LINKED ISSUE #46: Increase the tests coverage
## Issue Description
Increase the tests coverage

## Additional context
Run the two commands
```bash
$ docker-compose run api coverage run -m pytest
```
```bash
$ docker-compose run api coverage report -m
```
And see the report to know where need more tests to increase coverage.

### PR DESCRIPTION
**Pull request by lboaventura25**
## Issue 
Resolves #46 

## Description
Add more tests and cases of tests.

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

## How to validate?
Run the tests with
```bash
$ docker-compose run api coverage run -m pytest
```
And check the coverage with command
```bash
$ docker-compose run api coverage report -m
```

### CODE REVIEW

File: src/controllers/neighborhood.py [hunk 1]
```diff
@@ -28,25 +28,24 @@ def get_all_neighborhoods(city=None, state=None):
...
-                         }
-                ratings, status = db.get_all(Rating, filter)
-                ratings = [get_row_dict(rat) for rat in ratings]
-                if (ratings):
-                    statistics = get_neighborhood_statistics(ratings)
-                    neighborhoods[i].update(statistics)
 
-            return neighborhoods, code
-        return result, code
-    return [], 200
```
Comment (silvasara) [inline src/controllers/neighborhood.py:49]: it is necessary to return to an empty list with code 200 when there is no record in the bank and when there was no error

File: src/controllers/occurrence.py [hunk 2]
```diff
@@ -57,13 +57,12 @@ def get_all_occurrences(user=None, occurrence_type=None):
         filter = {'user': [user]}
 
     result, code = db.get_all(Occurrence, filter)
-    if result:
-        if code == 200:  # if successful, returns the data
-            # converts rows to dict
-            occurrences = [occurrence.to_dict() for occurrence in result]
-            return occurrences, code
-        return result, code  # else, returns database error and error code
-    return [], 200
```
Comment (silvasara) [inline src/controllers/occurrence.py:66]: it is necessary to return to an empty list with code 200 when there is no record in the bank and when there was no error

File: src/controllers/user.py [hunk 3]
```diff
@@ -22,14 +22,17 @@ def create_user(body):
...
+        return result, code
+
     result, code = db.get_all(User)
-    if result:
-        if code == 200:  # if successful, returns the data
-            users = [u.to_dict() for u in result]  # converts rows to dict
-            return users, code
-        return result, code  # else, returns database error and error code
-    return [], 200
+
```
Comment (silvasara) [inline src/controllers/user.py:31]: it is necessary to return to an empty list with code 200 when there is no record in the bank and when there was no error

File: src/controllers/user.py [hunk 4]
```diff
@@ -22,14 +22,17 @@ def create_user(body):
     return result, code
 
 
-def get_all_users():
+def get_all_users(filter=None):
+    if filter:
+        result, code = db.get_all(User, {"invalid": 5})
+        return result, code
```
Comment (silvasara) [inline src/controllers/user.py:28]: If we don't have any filter functionality for user, this need not be done. There is no need to test functionality that does not exist

File: src/tests/test_config.py [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+
+class TestSettings(unittest.TestCase):
+    def test_load_configuration(self):
+        """
+        Testing load configuration function
+        """
+        config = load_configuration()
+        self.assertEqual(config['debug'], True)
+        self.assertEqual(config['host'], '0.0.0.0')
+        self.assertEqual(config['port'], 8083)
```
Comment (silvasara) [inline src/tests/test_config.py:13]: Notes that these settings can be obtained through environment variables, depending on how the deployment is done in the production environment, these variables can change and then these tests will break. I prefer that don't have these tests.

File: src/tests/test_user.py [hunk 6]
```diff
@@ -5,7 +5,8 @@
     correct_users,
     wrong_users,
     correct_user_update,
-    wrong_user_update
+    wrong_user_update,
```
Comment (silvasara) [inline src/tests/test_user.py:8]: This is not being used

**Review by silvasara — CHANGES_REQUESTED** [review]

File: src/controllers/user.py [hunk 7]
```diff
@@ -22,14 +22,17 @@ def create_user(body):
...
+
```
Comment (silvasara) [inline src/controllers/user.py:29]: this is not necessary as we do not have a user filter.  don't need to test features that we don't have

**Review by silvasara — CHANGES_REQUESTED** [review]

**Review by silvasara — APPROVED** [review]

**Review by renan601 — APPROVED** [review]

**Pull request merged**